[mfx-dispatch] Build error on x64-linux

Package: mfx-dispatch:x64-linux@1.35.1#4

**Host Environment**

- Host: x64-linux
- Compiler: GNU 14.2.1
-    vcpkg-tool version: 2025-03-22-d4dd14112c2026ee57e59f966ecfa3b9137bd143
    vcpkg-scripts version: 64f3d3d620 2025-03-27 (4 hours ago)

**To Reproduce**

`vcpkg install mfx-dispatch`

**Failure logs**

```
-- Using cached lu-zero-mfx_dispatch-1.35.1.tar.gz
-- Cleaning sources at /home/thomas/vcpkg/buildtrees/mfx-dispatch/src/1.35.1-3e0a7e503c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/thomas/vcpkg/downloads/lu-zero-mfx_dispatch-1.35.1.tar.gz
-- Applying patch fix-unresolved-symbol.patch
-- Applying patch fix-pkgconf.patch
-- Using source at /home/thomas/vcpkg/buildtrees/mfx-dispatch/src/1.35.1-3e0a7e503c.clean
-- Found external ninja('1.12.1').
-- Getting CMake variables for x64-linux-dbg
-- Getting CMake variables for x64-linux-rel
-- Generating configure for x64-linux
-- Finished generating configure for x64-linux
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/make V=1 -j 17 -f Makefile all
    Working Directory: /home/thomas/vcpkg/buildtrees/mfx-dispatch/x64-linux-dbg/
    See logs for more information:
      /home/thomas/vcpkg/buildtrees/mfx-dispatch/build-x64-linux-dbg-out.log
      /home/thomas/vcpkg/buildtrees/mfx-dispatch/build-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:136 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:2 (vcpkg_build_make)
  ports/mfx-dispatch/portfile.cmake:26 (vcpkg_install_make)
  scripts/ports.cmake:203 (include)



```

<details><summary>/home/thomas/vcpkg/buildtrees/mfx-dispatch/build-x64-linux-dbg-err.log</summary>

```
./../src/1.35.1-3e0a7e503c.clean/src/mfxloader.cpp:82:9: warning: "VERSION" redefined
   82 | #define VERSION(major, minor) {{minor, major}}
      |         ^~~~~~~
<command-line>: note: this is the location of the previous definition
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp: In function ‘std::string MFX::printCodecId(mfxU32)’:
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:60:3: error: ‘uint8_t’ was not declared in this scope
   60 |   uint8_t* data = reinterpret_cast<uint8_t*>(&id);
      |   ^~~~~~~
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:29:1: note: ‘uint8_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
   28 | #include "mfxloader.h"
  +++ |+#include <cstdint>
   29 | 
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:60:12: error: ‘data’ was not declared in this scope; did you mean ‘std::data’?
   60 |   uint8_t* data = reinterpret_cast<uint8_t*>(&id);
      |            ^~~~
      |            std::data
In file included from /usr/include/c++/14/list:64,
                 from ./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:26:
/usr/include/c++/14/bits/range_access.h:344:5: note: ‘std::data’ declared here
  344 |     data(initializer_list<_Tp> __il) noexcept
      |     ^~~~
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:60:36: error: ‘uint8_t’ does not name a type
   60 |   uint8_t* data = reinterpret_cast<uint8_t*>(&id);
      |                                    ^~~~~~~
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:60:36: note: ‘uint8_t’ is defined in header ‘<cstdint>’; this is probably fixable by adding ‘#include <cstdint>’
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:60:43: error: expected ‘>’ before ‘*’ token
   60 |   uint8_t* data = reinterpret_cast<uint8_t*>(&id);
      |                                           ^
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:60:43: error: expected ‘(’ before ‘*’ token
   60 |   uint8_t* data = reinterpret_cast<uint8_t*>(&id);
      |                                           ^
      |                                           (
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:60:44: error: expected primary-expression before ‘>’ token
   60 |   uint8_t* data = reinterpret_cast<uint8_t*>(&id);
      |                                            ^
./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp:60:50: error: expected ‘)’ before ‘;’ token
   60 |   uint8_t* data = reinterpret_cast<uint8_t*>(&id);
      |                                                  ^
      |                                                  )
make: *** [Makefile:722: src/libmfx_la-mfxparser.lo] Error 1
make: *** Waiting for unfinished jobs....
```
</details>

<details><summary>/home/thomas/vcpkg/buildtrees/mfx-dispatch/build-x64-linux-dbg-out.log</summary>

```
/bin/sh ./libtool  --tag=CXX   --mode=compile g++ -DPACKAGE_NAME=\"libmfx\" -DPACKAGE_TARNAME=\"libmfx\" -DPACKAGE_VERSION=\"1.34\" -DPACKAGE_STRING=\"libmfx\ 1.34\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libmfx\" -DVERSION=\"1.34\" -DMFX_HAVE_LINUX=1 -DHAVE_LIBVA_DRM=1 -DHAVE_LIBVA_X11=1 -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I. -I./../src/1.35.1-3e0a7e503c.clean    -I./../src/1.35.1-3e0a7e503c.clean/include -DMINGW_HAS_SECURE_API=1 -DMFX_MODULES_DIR=\".\" -DMFX_PLUGINS_CONF_DIR=\".\"   -fPIC -g -fPIC -MT src/libmfx_la-mfxloader.lo -MD -MP -MF src/.deps/libmfx_la-mfxloader.Tpo -c -o src/libmfx_la-mfxloader.lo `test -f 'src/mfxloader.cpp' || echo './../src/1.35.1-3e0a7e503c.clean/'`src/mfxloader.cpp
/bin/sh ./libtool  --tag=CXX   --mode=compile g++ -DPACKAGE_NAME=\"libmfx\" -DPACKAGE_TARNAME=\"libmfx\" -DPACKAGE_VERSION=\"1.34\" -DPACKAGE_STRING=\"libmfx\ 1.34\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libmfx\" -DVERSION=\"1.34\" -DMFX_HAVE_LINUX=1 -DHAVE_LIBVA_DRM=1 -DHAVE_LIBVA_X11=1 -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I. -I./../src/1.35.1-3e0a7e503c.clean    -I./../src/1.35.1-3e0a7e503c.clean/include -DMINGW_HAS_SECURE_API=1 -DMFX_MODULES_DIR=\".\" -DMFX_PLUGINS_CONF_DIR=\".\"   -fPIC -g -fPIC -MT src/libmfx_la-mfxparser.lo -MD -MP -MF src/.deps/libmfx_la-mfxparser.Tpo -c -o src/libmfx_la-mfxparser.lo `test -f 'src/mfxparser.cpp' || echo './../src/1.35.1-3e0a7e503c.clean/'`src/mfxparser.cpp
libtool: compile:  g++ -DPACKAGE_NAME=\"libmfx\" -DPACKAGE_TARNAME=\"libmfx\" -DPACKAGE_VERSION=\"1.34\" "-DPACKAGE_STRING=\"libmfx 1.34\"" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libmfx\" -DVERSION=\"1.34\" -DMFX_HAVE_LINUX=1 -DHAVE_LIBVA_DRM=1 -DHAVE_LIBVA_X11=1 -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I. -I./../src/1.35.1-3e0a7e503c.clean -I./../src/1.35.1-3e0a7e503c.clean/include -DMINGW_HAS_SECURE_API=1 -DMFX_MODULES_DIR=\".\" -DMFX_PLUGINS_CONF_DIR=\".\" -fPIC -g -fPIC -MT src/libmfx_la-mfxparser.lo -MD -MP -MF src/.deps/libmfx_la-mfxparser.Tpo -c ./../src/1.35.1-3e0a7e503c.clean/src/mfxparser.cpp -o src/libmfx_la-mfxparser.o
libtool: compile:  g++ -DPACKAGE_NAME=\"libmfx\" -DPACKAGE_TARNAME=\"libmfx\" -DPACKAGE_VERSION=\"1.34\" "-DPACKAGE_STRING=\"libmfx 1.34\"" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DPACKAGE=\"libmfx\" -DVERSION=\"1.34\" -DMFX_HAVE_LINUX=1 -DHAVE_LIBVA_DRM=1 -DHAVE_LIBVA_X11=1 -DHAVE_STDIO_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_STRINGS_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_UNISTD_H=1 -DSTDC_HEADERS=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -I. -I./../src/1.35.1-3e0a7e503c.clean -I./../src/1.35.1-3e0a7e503c.clean/include -DMINGW_HAS_SECURE_API=1 -DMFX_MODULES_DIR=\".\" -DMFX_PLUGINS_CONF_DIR=\".\" -fPIC -g -fPIC -MT src/libmfx_la-mfxloader.lo -MD -MP -MF src/.deps/libmfx_la-mfxloader.Tpo -c ./../src/1.35.1-3e0a7e503c.clean/src/mfxloader.cpp -o src/libmfx_la-mfxloader.o
mv -f src/.deps/libmfx_la-mfxloader.Tpo src/.deps/libmfx_la-mfxloader.Plo
```
</details>

--- a/ports/mfx-dispatch/portfile.cmake
+++ b/ports/mfx-dispatch/portfile.cmake
@@ -1,3 +1,10 @@
+vcpkg_download_distfile(
+    MISSING_CSTDINT_IMPORT_PATCH
+    URLS https://github.com/lu-zero/mfx_dispatch/commit/d6241243f85a0d947bdfe813006686a930edef24.patch?full_index=1
+    FILENAME fix-missing-cstdint-import-d6241243f85a0d947bdfe813006686a930edef24.patch
+    SHA512 5d2ffc4ec2ba0e5859d01d2e072f75436ebc3e62e0f6580b5bb8b9f82fe588e7558a46a1fdfa0297a782c0eeb8f50322258d0dd9e41d927cc9be496727b61e44
+)
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO lu-zero/mfx_dispatch
@@ -7,6 +14,7 @@ vcpkg_from_github(
     PATCHES
         fix-unresolved-symbol.patch
         fix-pkgconf.patch
+        ${MISSING_CSTDINT_IMPORT_PATCH}
 )
 
 if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)

--- a/ports/mfx-dispatch/vcpkg.json
+++ b/ports/mfx-dispatch/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "mfx-dispatch",
   "version": "1.35.1",
-  "port-version": 4,
+  "port-version": 5,
   "description": "Open source Intel media sdk dispatcher",
   "homepage": "https://github.com/lu-zero/mfx_dispatch",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6014,7 +6014,7 @@
     },
     "mfx-dispatch": {
       "baseline": "1.35.1",
-      "port-version": 4
+      "port-version": 5
     },
     "mgnlibs": {
       "baseline": "2019-09-29",

--- a/versions/m-/mfx-dispatch.json
+++ b/versions/m-/mfx-dispatch.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "0001879960211c1331aa7b26c40e0fdd36cf5f47",
+      "version": "1.35.1",
+      "port-version": 5
+    },
+    {
       "git-tree": "0e2d18fc6010dd23044bade1a855669f8e9c9c86",
       "version": "1.35.1",
       "port-version": 4

